modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/BackgroundTaskExecutorConfigurator.java

```diff
@@ -35,22 +41,46 @@ protected void activate(BundleContext bundleContext) {
 				bundleContext, _concurrentReindexManagerSnapshot.get(),
 				_portalExecutorManager, _syncReindexManagerSnapshot.get());
 
+		systemIndexers = ServiceTrackerListFactory.open(
```

Comment:
@rcsdnj Please keep `systemIndexers` in `ReindexSingleIndexerBackgroundTaskExecutor`, and expose a method to call `close()` on it.

In `deactivate()` method of this class, after unregistering, call that method.